docs/in-depth/configuration-file.md

```diff
@@ -53,16 +53,20 @@ A full list of modules available out-of-the-box, along with documentation about
 
 ## Declaring input particles
 
-Particles passed as input to the `computeWeights()` function (see [Calling MoMEMta](calling-momemta)) can be linked with the modules by registering them in the Lua configuration. For instance, for a particle declared in C++ as `momemta::Particle m_part1 { "part1", p1, 0 }`, one would do:
+Input particles are declared in the configuration file as:
 ```Lua
-local l_part1 = declare_input("part1") -- Pass as argument the name string of m_part1
+local l_part1 = declare_input("part1") -- Give that input a name: "part1"
 ```
-The Lua object `l_part1` now contains the input tag needed to access the particle's 4-vector. For instance, defining a transfer function on the energy of `part1` can be done as:
+
+In C++, that input's 4-vector can then be passed through the `computeWeights()` function (see [Calling MoMEMta](calling-momemta)) by defining a `momemta::Particle` object whose name attribute is the same as the one used in the configuration, e.g. `momemta::Particle m_part1 { "part1", p1, 0 }`.
+
+The Lua object `l_part1` contains the input tag needed to access the particle's 4-vector. For instance, defining a transfer function on the energy of `part1` can be done as:
```

Comment:
maybe adding something like `via the `reco_p4` attribute` at the end of the first sentence?